### PR TITLE
fix: retry PDML on Aborted and Internal errors

### LINKED ISSUE #1197: spanner-client: Retry PDML on "Received unexpected EOS on DATA frame from server"
This bug is related to the Spanner client library.

For long lived transactions (>= 30 minutes), in the case of large PDML changes, it is possible that the gRPC connection is terminated with an error "Received unexpected EOS on DATA frame from server".

In this case, we need to retry the transaction either with the received resume token obtained on reading the stream or from scratch. This will ensure that the PDML transaction continues to execute until it is successful or a hard timeout is reached.

We have already implemented such change in the Java client library, for more information see this PR: https://github.com/googleapis/java-spanner/pull/360.

In order to test the fix, we can use a large spanner database. Please speak to @thiagotnunes for more details.

### PR DESCRIPTION
**Pull request by olavloite**
PDML statements should be retried on the following errors:
* ABORTED: Retry using a new transaction
* INTERNAL(Received unexpected EOS on DATA frame from server):
    Retry using the same transaction and any resume token that was received.

Fixes #1197, #1177

### CODE REVIEW

**Review by thiagotnunes — COMMENTED** [review]
LGTM, but we will need to internally test it to make sure. I will try and do that next week. Thanks!

File: test/database.ts [hunk 1]
```diff
@@ -2235,7 +2235,11 @@ describe('Database', () => {
         'begin'
       ) as sinon.SinonStub).callsFake(callback => callback(null));
 
-      runUpdateStub = sandbox.stub(fakePartitionedDml, 'runUpdate');
+      // runUpdateStub = sandbox.stub(fakePartitionedDml, 'runUpdate');
```
Comment (skuruppu) [inline test/database.ts:2238]: nit: remove commented out line.

**Review by skuruppu — APPROVED** [review]

**Comment by skuruppu** [comment]
Adding a "do not merge" until @thiagotnunes tests the implementation.

**Comment by thiagotnunes** [comment]
@olavloite Still getting an error from the backend (a different one though): `ERROR: Error: 13 INTERNAL: Received RST_STREAM with code 2 (Internal server error)`

If we change the `isRetryable` function to something like, it should be enough:
```js
function isRetryableInternalError(err: grpc.ServiceError): boolean {
  return (
    err.code === grpc.status.INTERNAL &&
    (err.message.includes('Received unexpected EOS on DATA frame from server') || err.message.includes('Received RST_STREAM'))
  );
}
```

**Comment by olavloite** [comment]
@thiagotnunes Thanks a lot for testing it. I've updated the check to include this error as well.

I also had a quick look at a couple of the other client libraries:
1. In Go we also retry this error: https://github.com/googleapis/google-cloud-go/blob/14d43f9dd504466f50d8764d7dd346dce29ed98e/spanner/retry.go#L63
1. In Java we do not: https://github.com/googleapis/java-spanner/blob/6f47b8a759643f772230df0c2e153338d44f70ce/google-cloud-spanner/src/main/java/com/google/cloud/spanner/IsRetryableInternalError.java#L35

Could it be that this is not necessary in Java, because we set a keep-alive on the gRPC transport channel in Java: https://github.com/googleapis/java-spanner/blob/6f47b8a759643f772230df0c2e153338d44f70ce/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java#L316

**Comment by thiagotnunes** [comment]
@olavloite thanks for the fix!

I don't know why this is not happening in Java, but you might have a good lead there. By searching internally, it seems that for other clients have decided to also add this error as retryable in any case. I think we should add that in the java client to be on the safe side.

**Review by thiagotnunes — APPROVED** [review]

**Pull request merged**